### DISCUSSION TITLE
error type inference in queryClientConfig

### DISCUSSION THREAD

**Discussion by ariariel** [trpc/trpc#1648] [Q&A] [answered]
Hi! I have a quick question about type inference in regards to the custom errors defined in `formatError`. I've noticed that the types are inferred nicely when dealing with a specific query or mutation. However while configuring the query client in the HOC, the `onError` handler still thinks `error` is `unknown`.

```
queryClientConfig: {
  defaultOptions: {
    queries: {
      onError: err => {
        // err: unknown
      },
    },
  },
},
```

We are trying to wire up error reporting (to Sentry) at the top-level, instead of doing it per procedure call. It would be great to have the types inferred here too if possible. Or is there a better recommended way of dealing with tRPC errors application-wide?

**Comment by KATT** [comment, marked as answer]
You can typecast it as `TRPCClientErrorLike<AppRouter>`... feel free to open a PR or an issue to work on this. Happy to pay someone to fix it

**Reply by KATT** [reply]
I think my reasoning was that it's not _guaranteed_ to be a `TRPCClientError` because it's possible to use `react-query` without the trpc calls

**Reply by ariariel** [reply]
Thanks @KATT, I think your reasoning is right. We use `react-query` for non-TRPC purposes too and we may have errors that aren't TRPC errors. I've gone with the typecasting you suggested.  👍
